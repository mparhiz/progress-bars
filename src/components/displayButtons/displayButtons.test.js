import React from 'react';
import DisplayButtons from './DisplayButtons';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from "enzyme-to-json";
import * as enzyme from "enzyme";

enzyme.configure({ adapter: new Adapter() });

test('renders without crashing: Enzyme', () => {
  const wrapper = enzyme.mount(
    <DisplayButtons 
        buttons={[10,20]}
        applyValue={jest.fn()}
        dropdownItems={[0,1]}
        selectItem={jest.fn()}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('renders a dropdown', () => {
  const component = enzyme.mount(
    <DisplayButtons 
      buttons={[10,20]}
      applyValue={jest.fn()}
      dropdownItems={[0,1]}
      selectItem={jest.fn()}
    />
  );
  expect(component.exists('DropdownButton')).toEqual(true);
});

test('should call after clicking on first button', () => {
  const spy = jest.fn();
  const component = enzyme.mount(
    <DisplayButtons 
      buttons={[10,20]}
      applyValue={spy}
      dropdownItems={[0,1]}
      selectItem={jest.fn()}
    />
  );
  const firstButton = component.find('button').at(1);
  firstButton.simulate('click');

  expect(spy).toHaveBeenCalled();
});
