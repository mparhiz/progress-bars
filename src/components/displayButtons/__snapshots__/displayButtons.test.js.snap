// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing: Enzyme 1`] = `
<DisplayButtons
  applyValue={[MockFunction]}
  buttons={
    Array [
      10,
      20,
    ]
  }
  dropdownItems={
    Array [
      0,
      1,
    ]
  }
  selectItem={[MockFunction]}
>
  <div>
    <ButtonGroup
      className="mb-2"
      role="group"
      toggle={false}
      vertical={false}
    >
      <div
        className="mb-2 btn-group"
        role="group"
      >
        <DropdownButton
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "role": "group",
                "toggle": false,
                "vertical": false,
              },
              "render": [Function],
            }
          }
          id="bg-nested-dropdown"
          onSelect={[Function]}
          title="Progress bar 1"
        >
          <Dropdown
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "role": "group",
                  "toggle": false,
                  "vertical": false,
                },
                "render": [Function],
              }
            }
            navbar={false}
            onSelect={[Function]}
          >
            <ReactOverlaysDropdown
              itemSelector=".dropdown-item:not(.disabled):not(:disabled)"
              onToggle={[Function]}
            >
              <ButtonGroup
                className="dropdown"
                onKeyDown={[Function]}
                role="group"
                toggle={false}
                vertical={false}
              >
                <div
                  className="dropdown btn-group"
                  onKeyDown={[Function]}
                  role="group"
                >
                  <DropdownToggle
                    id="bg-nested-dropdown"
                  >
                    <Button
                      active={false}
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="dropdown-toggle"
                      disabled={false}
                      id="bg-nested-dropdown"
                      onClick={[Function]}
                      variant="primary"
                    >
                      <button
                        aria-expanded={false}
                        aria-haspopup={true}
                        className="dropdown-toggle btn btn-primary"
                        disabled={false}
                        id="bg-nested-dropdown"
                        onClick={[Function]}
                        type="button"
                      >
                        Progress bar 1
                      </button>
                    </Button>
                  </DropdownToggle>
                  <DropdownMenu
                    align="left"
                    alignRight={false}
                    flip={true}
                  />
                </div>
              </ButtonGroup>
            </ReactOverlaysDropdown>
          </Dropdown>
        </DropdownButton>
        <Button
          active={false}
          disabled={false}
          key="0"
          onClick={[Function]}
          variant="primary"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            10
          </button>
        </Button>
        <Button
          active={false}
          disabled={false}
          key="1"
          onClick={[Function]}
          variant="primary"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            20
          </button>
        </Button>
      </div>
    </ButtonGroup>
  </div>
</DisplayButtons>
`;
